src/platform-aws.c

```diff
@@ -424,6 +424,25 @@ int platform_init(void)
 			goto exit;
 		}
 	}
+
+	/*
+	 * NCCL v2.19.3 reduced the chunk size used when running NVLS Tree
+	 * algorithm on greater than 4 nodes to 64KiB. This drastically impacted
+	 * performance on AWS (Ref: https://github.com/NVIDIA/nccl/pull/1112/
+	 * for some data). NCCL v2.20.3 has made this a tunable, so revert back
+	 * to using 512KiB which recovers from the regression. Setting this
+	 * unconditionally without relying on ncclGetVersion symbol being
+	 * available, since the parameter did not exist in versions prior to
+	 * v2.20.
+	 */
+	NCCL_OFI_INFO(NCCL_INIT | NCCL_NET, "Setting NCCL_NVLSTREE_MAX_CHUNKSIZE to 512KiB");
```

Comment:
Did/do we want to test how this impacts our other platforms (P3/P4/G*)? Or make this a P5-specific change?

Reply:
I haven't tested the original PR (https://github.com/NVIDIA/nccl/pull/1112/) on non P5 instances. Because it just reverted the NVLSTree max chunk size behaviour to how it was prior to NCCL v2.19.3-1 commit. Therefore, it is suppose to be safe.